packages/lodestar/src/api/impl/beacon/state/utils.ts

```diff
@@ -198,3 +218,38 @@ export function filterStateValidatorsByStatuses(
   }
   return responses;
 }
+
+/**
+ * Get the archived state nearest to `slot`.
+ */
+async function getNearestArchivedState(
+  config: IBeaconConfig,
+  db: IBeaconDb,
+  slot: Slot
+): Promise<CachedBeaconState<allForks.BeaconState>> {
+  const states = db.stateArchive.valuesStream({lte: slot, gt: 0, reverse: true});
+  const state = (await states[Symbol.asyncIterator]().next()).value as TreeBacked<allForks.BeaconState>;
+  return createCachedBeaconState(config, state);
+}
+
+async function getFinalizedState(
+  config: IBeaconConfig,
+  db: IBeaconDb,
+  forkChoice: IForkChoice,
+  slot: Slot
+): Promise<CachedBeaconState<allForks.BeaconState>> {
+  assert.lte(slot, forkChoice.getFinalizedCheckpoint().epoch * config.params.SLOTS_PER_EPOCH);
+  let state = await getNearestArchivedState(config, db, slot);
+
+  // process blocks up to the requested slot
+  for await (const block of db.blockArchive.valuesStream({gt: state.slot, lte: slot})) {
+    state = fast.fastStateTransition(state, block);
```

Comment:
by default, all verification flags are `true`. I suppose we only need to regen state so probably make them all `false`?